pyro/contrib/gp/models/gpr.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Variable
+import torch.nn as nn
+
+import pyro
+import pyro.distributions as dist
+
+
+class GPRegression(nn.Module):
+    """
+    Gaussian Process regression module.
+    """
+    def __init__(self, X, y, kernel, noise=torch.ones(1), priors={}):
```

Comment:
nit: It is bit dangerous to use mutable values for default arguments, since all instances then share these mutable values. Prefer `None` with `is None` checks:
```py
def __init__(self, X, y, kernel, noise=torch.ones(1), priors={}):
   if noise is None:
        noise = X.new([1])  # using .new() also ensure noise is on correct cuda device
   if priors is None:
        priors = {}
```